v2: Drop IE8-IE10 support.

icio opened the pull request:
Analysis shows these are, thankfully, unused.

#### Checklist

#### Before merging:

- [x] Retarget this branch to a new `v2` branch.
- [x] Ensure desired [v1 PRs](https://github.com/unravelin/ravelinjs/pulls?q=is%3Apr+is%3Aopen+base%3Av1) are merged
- [x] Rebase the `v2` branch on `v1`.

#### After merging:

- [ ] Change the repo's default branch to `v2`.
- [ ] Change RavelinJS integration test in core to target the `v2` branch.

icio committed:
core: Use substring instead of substr.

icio committed:
core: Remove XDomainRequest support.

icio committed:
test: Stop testing IE8-IE10.

icio committed:
test: Remove option rav:send:skipCrossDomainTest to skip test (only used by IE8-10).

icio committed:
test: Remove manual entropy generation.

IE8-10 didn't support window.crypto.getRandomValues and our library fell back
to collecting entropy from mouse/keyboard events. There was a risk it hadn't
collected enough entropy, so we'd jiggle the mouse until it worked. IE11
supports getRandomValues so we shouldn't run into this error any more.

icio committed:
doc: Remove IE8-10 mentions from readme.

icio committed:
core: Remove XDR mentions.

icio committed:
test: Remove randomness-generation skips from encrypt.card tests.

icio committed:
build: ie8=false.

icio committed:
test: Re-enable checkLeaks?

icio committed:
doc: Refer to karma.conf.cjs.

icio committed:
chore: Upgrade rollup to v3. Fixes #236

timphillips commented on package.json:
```diff
@@ -67,9 +68,8 @@
     "remark-cli": "11.0.0",
     "remark-toc": "8.0.1",
-    "rollup": "2.79.1",
+    "rollup": "3.23.1",
```
🎉

timphillips commented on README.md:
```diff
@@ -394,12 +392,6 @@ var action = fetch('https://api.ravelin.com/v2/checkout?score=true', {
-`window.crypto`](https://caniuse.com/cryptography) (including IE8-IE10) rely on
-a pseudo-random number generator based on collecting user events from the page
-and that if this generator has not collected enough events it may throw an
-exception when trying to encrypt.
```
There are also some comments about `window.crypto` under the Browser Compatibility section — should those be updated too? And will we be able to remove the pseudo-random number generator fallback, or is it still needed for some other browsers?

icio replied:
Removed in 7aed85cca47f0203fdfee088bbdaf381b9d1734c

icio replied:
> And will we be able to remove the pseudo-random number generator fallback, or is it still needed for some other browsers?

This fallback is buried in our vendored dependency, so not something I think we should remove. cmd+f "msCrypto":

https://github.com/unravelin/ravelinjs/blob/4e9ea863a92181771e99594dc203e510a25cd046/lib/encryption-vendored.js#L774-L777

timphillips reviewed (dismissed):
Excellent. I did a cursory review and the changes all look reasonable.

icio committed:
doc: Remove mention of window.crypto fallbacks.

icio committed:
test: Re-enable mocha checkLeaks.

icio committed:
test: Remove checkLeaks test.

We've seen failures in Chrome 109 on Mojave where the Core#send beforeEach hook
has errored with:

    global leak detected: ret_nodes

    https://automate.browserstack.com/dashboard/v2/builds/d02d09eba240bfe63d5d4a38186f8b263e51448b/sessions/cf7edd2c7efa5ea1d0b8d2267290fa7bdcb0595a?auth_token=e9574e9467e1f22131acef157981aeb0791f3016cc7568e5faf4fbbf550f6402&build_token=aWYzOUl2OVQxRVBrWURhSUdKZWliZU1TV01NTStuQlNUejdRSTE1djgrWXNXWUVRZmh0d25KM3lSSHVKaC91NmFRRjRreFN6RnhMWFNsOS9pb25IM0E9PS0tejEwVU1hSlNsSHJKczZDK0Y3YjlHdz09--d2cf7eb2a908ea8939f86035ad68730cc7ea2969

I've tried to track the source of ret_nodes, but been unable to find it in our
code, our vendored code, or the test-framework dependencies in
test/unit/index.html. I think we're fine without the check.

icio committed:
test: Remove encrypt test for seeded generators.

icio committed:
test: Upgrade mocha, jquery in unit tests.

expect is the latest version (last updated 9 years ago) and xhook 1.4 is the
last version to support IE11.

icio committed:
doc: Target v2 for dropping IE8-10 support.

icio committed:
2.0.0-beta.0

timphillips commented on CONTRIBUTING.md:
```diff
@@ -184,10 +184,12 @@ HTML file you write for you test, but most will use:
 
 The main branches of the ravelinjs repo follow their major semver version:
-[v1](https://github.com/unravelin/ravelinjs/tree/v1) (latest, default) and
+[v2](https://github.com/unravelin/ravelinjs/tree/v1) (latest, default - largely
+the same as v2 but without IE8-10 support),
```
```suggestion
[v2](https://github.com/unravelin/ravelinjs/tree/v2) (latest, default - largely
the same as v1 but without IE8-10 support),
```

timphillips commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
```
We should update `"branch": "v1",` in the `np` section too.

timphillips commented on test/unit/index.html:
```diff
@@ -4,19 +4,18 @@
 
     <script src="https://unpkg.com/xhook@1.4.9/dist/xhook.js"></script>
-    <script src="https://unpkg.com/jquery@1.12.4/dist/jquery.js"></script>
+    <script src="https://unpkg.com/jquery@2.2.4/dist/jquery.js"></script>
```
The jQuery version in [CONTRIBUTING.md](https://github.com/unravelin/ravelinjs/blob/3de07351cd98a7cd8aad9548dd315f89303be8af/CONTRIBUTING.md?plain=1#L111) can be bumped to v2.

timphillips reviewed (dismissed):
Looks good apart from a few minor things.

icio committed:
doc: Correct upgraded deps in CONTRIBUTING.md.

icio committed:
chore: np should now publish from v2.

icio committed:
Update CONTRIBUTING.md

Co-authored-by: Tim Phillips <tim.phillips@ravelin.com>

icio commented:
@timphillips You might need to do the honours from here :v:

robcrocombe commented:
Replaced with #487
